sse2neon.h

```diff
@@ -1529,6 +1529,21 @@ FORCE_INLINE __m128i _mm_srai_epi16(__m128i a, int count)
         ret;                                                            \
     })
 
+// Shifts the 8 signed or unsigned 16-bit integers in a left by count bits while
+// shifting in zeros.
+//
+//   r0 := a0 << count
+//   r1 := a1 << count
+//   ...
+//   r7 := a7 << count
+//
+// https://msdn.microsoft.com/en-us/library/c79w388h(v%3dvs.90).aspx
+FORCE_INLINE __m128i _mm_sll_epi16(__m128i a, __m128i count)
+{
+    int64_t c = ((SIMDVec *) &count)->m128_i64[0];
+    int16x8_t vc = vdupq_n_s16(c);
+    return vreinterpretq_m128i_s16(vshlq_s16(vreinterpretq_s16_m128i(a), vc));
+}
```

Comment:
Be aware that when `c` equals to `256`, the element value of `vc` vector is `1`.
Hence the returned vector would be shifted `1` bit instead of being set to `0`.

We might need a mask `0xFFF...F00` to check whether the count is too large or too small.
If it is, set all the element to 0.
Otherwise, shift as usual.

Reply:
I think the new patch fixes this problem.